pkg/operator/webhook_removal_controller.go

```diff
@@ -0,0 +1,121 @@
+package operator
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+)
+
+var (
+	csiSnapshotConditionPrefix       = "CSISnapshotWebhook"
+	webHookControllerConditionPrefix = "WebhookController"
+)
+
+type webhookRemovalController struct {
+	name                 string
+	manifests            resourceapply.AssetFunc
+	operatorClient       v1helpers.OperatorClient
+	guestCompositeClient *resourceapply.ClientHolder
+	mgmtCompositeClient  *resourceapply.ClientHolder
+	eventRecorder        events.Recorder
+}
+
+func NewWebhookRemovalController(
+	name string,
+	manifests resourceapply.AssetFunc,
+	operatorClient v1helpers.OperatorClient,
+	guestCompositeClient *resourceapply.ClientHolder,
+	mgmtCompositeClient *resourceapply.ClientHolder,
+	eventRecorder events.Recorder,
+) factory.Controller {
+	c := &webhookRemovalController{
+		name:                 name,
+		manifests:            manifests,
+		operatorClient:       operatorClient,
+		guestCompositeClient: guestCompositeClient,
+		mgmtCompositeClient:  mgmtCompositeClient,
+		eventRecorder:        eventRecorder,
+	}
+
+	return factory.New().WithInformers().
+		WithSync(c.sync).
+		ResyncEvery(time.Minute).
+		WithSyncDegradedOnError(operatorClient).
+		ToController(c.name, eventRecorder.WithComponentSuffix("webhook-removal-controller-"))
+}
+
+func (c *webhookRemovalController) sync(ctx context.Context, syncContext factory.SyncContext) error {
+	spec, status, _, err := c.operatorClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+
+	if spec.ManagementState != operatorv1.Managed {
+		return nil
+	}
+
+	guestAssetsTobeRemoved := []string{
+		"rbac/webhook_clusterrole.yaml",
+		"rbac/webhook_clusterrolebinding.yaml",
+		"webhook_config.yaml",
+	}
+	guestResourceRemovalResult := resourceapply.DeleteAll(ctx, c.guestCompositeClient, c.eventRecorder, c.manifests, guestAssetsTobeRemoved...)
+	allErrors := []error{}
+
+	for _, result := range guestResourceRemovalResult {
+		if result.Error != nil {
+			allErrors = append(allErrors, result.Error)
+		}
+	}
+
+	mgmtAssetsTobeRemoved := []string{
+		"webhook_serviceaccount.yaml",
+		"webhook_service.yaml",
+		"webhook_deployment_pdb.yaml",
+		"webhook_deployment.yaml",
+	}
+	mgmtResouceRemovalResult := resourceapply.DeleteAll(ctx, c.mgmtCompositeClient, c.eventRecorder, c.manifests, mgmtAssetsTobeRemoved...)
+	for _, result := range mgmtResouceRemovalResult {
+		if result.Error != nil {
+			allErrors = append(allErrors, result.Error)
+		}
+	}
+
+	if len(allErrors) > 0 {
+		return utilerrors.NewAggregate(allErrors)
+	}
```

Comment:
`NewAggregate` already filters `nil`. In theory, this could be enough:
```
guestResourceRemovalErr := resourceapply.DeleteAll()
mgmtResouceRemovalErr := resourceapply.DeleteAll()
if err := utilerrors.NewAggregate(append(guestResourceRemovalErr,mgmtResouceRemovalErr... ); err != nil {
     return err
}

Reply:
wait, I forgot. we will have to still extract errors from these objects in a loop. Result of `.DeleteAll` call is not of error type.